tests/e2e/GetMatchingContextsTest.php

```diff
@@ -1,40 +1,89 @@
 <?php
+declare(strict_types=1);
 
 namespace App\Tests\e2e;
 
+use App\Helper\ArrayHelper;
+
 class GetMatchingContextsTest extends BaseApiE2eTestCase
 {
-    public function getMatchingContextsData(): array
+    public function testGetUnfilteredMatchingContexts(): void
     {
-        return [
-            [null, 6, ['http://site-ecologique.fr', 'http://random-site.fr', 'http://site-ecologique-et-politique.fr', 'http://expired.fr', "(duckduckgo\.com|www\.bing\.com|www\.google\.fr|www\.qwant\.com|www\.yahoo\.com|first\.domainname\.fr|second\.domainname\.fr)/superexample", 'http://siteecologique.fr']],
-            [['john_doe', 'contributor2'], 5, ['http://site-ecologique.fr', 'http://site-ecologique-et-politique.fr', 'http://random-site.fr', 'http://expired.fr', 'http://siteecologique.fr']],
-            [['john_doe'], 3, ['http://site-ecologique.fr', 'http://random-site.fr', 'http://siteecologique.fr']],
-            [['contributor2'], 2, ['http://site-ecologique-et-politique.fr', 'http://expired.fr']],
-        ];
+        $payload = $this->makeApiRequest('/api/v3/matchingcontexts');
+
+        self::assertCount(8, $payload);
+
+        $this->assertMatchingContextsAllHaveValidNoticeUrls($payload);
+
+        self::assertEqualsCanonicalizing([
+            '(dismoi\.io)/aide',
+            '(dismoi\.io)/aide',
+            'http://site-ecologique.fr',
+            'http://random-site.fr',
+            'http://site-ecologique-et-politique.fr',
+            'http://expired.fr',
+            "(duckduckgo\.com|www\.bing\.com|www\.google\.fr|www\.qwant\.com|www\.yahoo\.com|first\.domainname\.fr|second\.domainname\.fr)/superexample",
+            'http://siteecologique.fr'
+        ], $this->pluckUrlRegexes($payload));
+
+        $matchingXPath = $this->referenceRepository->getReference('mc_matching_xpath');
+        $aideDisMoi = ArrayHelper::find($payload, function ($matchingContext) use ($matchingXPath) {
+            return $matchingContext['id'] === $matchingXPath->getId();
+        });
+        self::assertEquals("//text()[contains(.,'Aide')]", $aideDisMoi['xpath']);
```

Comment:
I find these kind of tests too specific and hard to maintain. Maybe something testing only the shape of the payload would be better?

Reply:
OK … I wanted to keep the same granularity as the existing one … Let me try something lighter

Reply:
Done. Tell me what you think …

Reply:
Didn't see any significant changes, maybe something left to push? :sweat_smile:

Reply:
No more count, no more reference comparison … Are you looking at the up-to-date diff @lutangar  ?

![image](https://user-images.githubusercontent.com/3037833/100346328-f043f800-2fe3-11eb-869d-2c8b040ae61e.png)